includes/wc-template-functions.php

```diff
@@ -1966,7 +1966,7 @@ function woocommerce_form_field( $key, $args, $value = null ) {
 
 					$field .= '<input type="hidden" class="hidden" name="' . esc_attr( $key ) . '" id="' . esc_attr( $args['id'] ) . '" value="" ' . implode( ' ', $custom_attributes ) . ' placeholder="' . esc_attr( $args['placeholder'] ) . '" />';
 
-				} elseif ( is_array( $states ) ) {
+				} elseif ( is_array(!empty( $states )) ) {
```

Comment:
No idea for what issue.
And if is a different issue, so should be into a different Pull Request.